ci/Jenkinsfile

```diff
@@ -1,39 +1,134 @@
-pipeline {
-    agent { label 'permaplant' }
 
-    stages {
-        stage('Build PR') {
-            environment {
-                DATABASE_URL = 'postgres://cidb:cidb@127.0.0.1/cidb'
+stage('Build Schema'){
+    node('docker') {
+        checkout scm
+        def rust_image = docker.build("permaplant-rust", "./ci/container-images/permaplant-rust")
+
+        docker.image('postgres:alpine').withRun('-e "POSTGRES_USER=ci" -e "POSTGRES_PASSWORD=ci"') { c ->
+            rust_image.inside("--link ${c.id}:db -e 'DATABASE_URL=postgres://ci:ci@db/ci'") {
+                checkout scm
+                sh './ci/build-scripts/build-schema.sh'
+
+                stash includes: 'backend/src/schema.rs', name: 'schema.rs'
+                stash includes: 'frontend/src/bindings/definitions.ts', name: 'definitions.ts'
             }
+        }
+    }
+}
 
-            steps {
-                echo "Workspace is '${WORKSPACE}'"
+stage('Checks') {
+    def parallel_checks = [:]
 
-                echo 'Ensure pr service is stopped'
-                sh 'sudo /usr/sbin/service permaplant-pr stop'
+    parallel_checks["check-backend"] = {
+        node('docker') {
+            checkout scm
+            def rust_image = docker.build("permaplant-rust", "./ci/container-images/permaplant-rust")
 
-                echo 'Resetting the CI DB ...'
-                sh 'sudo /usr/local/bin/permaplant-reset-ci.sh'
+            docker.image('postgres:alpine').withRun('-e "POSTGRES_USER=ci" -e "POSTGRES_PASSWORD=ci"') { c ->
+                rust_image.inside("--link ${c.id}:db -e 'DATABASE_URL=postgres://ci:ci@db/ci'") {
+                    checkout scm
+                    sh './ci/build-scripts/build-schema.sh'
 
-                echo 'Building...'
-                sh './ci/build-scripts/build-backend.sh'
-                sh './ci/build-scripts/build-frontend.sh'
+                    dir("backend") {
+                        withEnv(['RUSTFLAGS=-D warnings', 'RUSTDOCFLAGS=-D warnings']) {
+                            sh "cargo fmt --check"
+                            sh "cargo check"
+                            sh "cargo clippy"
+                            sh "cargo doc"
```

Comment:
Is this fast enough to run this serial?

Reply:
I specifically chose the commands in this order so they fails as fast as possible. Meaning `cargo check` runs faster than `cargo doc` and that is why its first.

Apart from that when we run the commands in parallel stages each of them has to build all dependencies of the project. This takes a lot of time and resources therefore in my opinion it is better to run them sequentially (so we only have to build dependencies once).

/cc @lukashartl